script/copy_deduplicate

```diff
@@ -266,49 +270,85 @@ def copy_join_parts(client, stable_table, query_jobs):
             print(f"Deleted {len(query_jobs)} temporary tables")
 
 
+def contains_glob(patterns):
+    return any(set("*?[").intersection(pattern) for pattern in patterns)
+
+
+def glob_dataset(pattern):
+    return pattern.split(".", 1)[0]
+
+
+def compile_glob_patterns(patterns):
+    return re.compile("|".join(fnmatch.translate(pattern) for pattern in patterns))
+
+
+def glob_predicate(match, table, arg):
+    matched = match(table) is not None
+    if (arg == "only") ^ matched:
```

Comment:
Can this be expressed without `^`, or can we add a comment to explain what that operator is doing here?

Reply:
sure, i'll change it to:
```suggestion
    if (arg == "only" and not matched) or (arg == "except" and matched):
```